drivers/usb/device/usb_dc_stm32.c

```diff
@@ -254,6 +254,25 @@ static int usb_dc_stm32_clock_enable(void)
 		LOG_ERR("Unable to set USB clock source to PLL.");
 	}
 #endif /* CONFIG_CLOCK_STM32_MSI_PLL_MODE && !CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI */
+
+#elif defined(RCC_CFGR_OTGFSPRE) \
+	&& defined(LL_RCC_USB_CLKSOURCE_PLL_DIV_2) \
+	&& defined(LL_RCC_USB_CLKSOURCE_PLL_DIV_3)
+
+	/* the USB OTGFSCLK needs to be 48 MHz, options are:
+	 *   - PLLCLK * 2 / 2
+	 *   - PLLCLK * 2 / 3
+	 *
+	 * this presumes that the system is running from PLLCLK
+	 */
+	if (SystemCoreClock == 48000000U) {
```

Comment:
As mentioned, we're trying to get rid of `SystemCoreClock` variable which is available only on ARM platform.
Please use `sys_clock_hw_cycles_per_sec`.